src/mahoji/lib/abstracted_commands/maniacalMonkeyCommand.ts

```diff
@@ -0,0 +1,315 @@
+	timeToFinish -= (timeToFinish * percentReduced) / 100;
+
+	const [newTime, skillBoostMsg] = applySkillBoost(user, timeToFinish, attackStyles);
+
+	timeToFinish = newTime;
+	boosts.push(skillBoostMsg);
+
+	if (percentReduced >= 1) boosts.push(`${percentReduced}% for KC`);
+
+	// Initialize consumable costs before any are calculated.
+	const consumableCosts: Consumable[] = [];
+
+	// Calculate Chinning and Barrage boosts + costs:
+	if (method === 'barrage' && attackStyles.includes(SkillsEnum.Magic)) {
+		if (user.skillLevel(SkillsEnum.Magic) < 94) {
+			return `You need 94 Magic to use Ice Barrage. You have ${user.skillLevel(SkillsEnum.Magic)}`;
+		}
+		if (user.hasEquippedOrInBank('Occult necklace')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Occult necklace');
+		}
+		if (user.hasEquippedOrInBank('Tormented bracelet')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Tormented bracelet');
+		}
+		if (user.hasEquippedOrInBank('Kodai wand')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Kodai wand');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe top')) {
+			timeToFinish *= (100 - 3) / 100;
+			boosts.push('3% for Ancestral robe top');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe bottom')) {
+			timeToFinish *= (100 - 2) / 100;
+			boosts.push('2% for Ancestral robe bottom');
+		}
+		consumableCosts.push(iceBarrageConsumables);
+		timeToFinish = reduceNumByPercent(timeToFinish, 75);
+		boosts.push('75% for Ice Barrage');
+	} else if (method === 'burst' && attackStyles.includes(SkillsEnum.Magic)) {
+		if (user.skillLevel(SkillsEnum.Magic) < 70) {
+			return `You need 70 Magic to use Ice Burst. You have ${user.skillLevel(SkillsEnum.Magic)}`;
+		}
+		if (user.hasEquippedOrInBank('Occult necklace')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Occult necklace');
+		}
+		if (user.hasEquippedOrInBank('Tormented bracelet')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Tormented bracelet');
+		}
+		if (user.hasEquippedOrInBank('Kodai wand')) {
+			timeToFinish *= (100 - 5) / 100;
+			boosts.push('5% for Kodai wand');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe top')) {
+			timeToFinish *= (100 - 3) / 100;
+			boosts.push('3% for Ancestral robe top');
+		}
+		if (user.hasEquippedOrInBank('Ancestral robe bottom')) {
+			timeToFinish *= (100 - 2) / 100;
+			boosts.push('2% for Ancestral robe bottom');
+		}
+		consumableCosts.push(iceBurstConsumables);
+		timeToFinish = reduceNumByPercent(timeToFinish, 67);
+		boosts.push('67% for Ice Burst');
+	} else if (method === 'chinning' && attackStyles.includes(SkillsEnum.Ranged)) {
+		// Check what Chinchompa to use
+		const chinchompas = ['Black chinchompa', 'Red chinchompa', 'Chinchompa'];
+		let chinchompa = 'Black chinchompa';
+		for (let chin of chinchompas) {
+			if (user.owns(chin) && user.bank.amount(chin) > 5000) {
+				chinchompa = chin;
+				break;
+			}
+		}
+		const regularVoid = resolveItems([
```

Comment:
Since this is already a thing in TOB code would it be more beneficial to create it as an export somewhere? Just helps any code relying on void in the future. I know it may go outside of the scope of this pr